dvc/remote/base.py

```diff
@@ -547,11 +547,49 @@ def download(
         if to_info.scheme != "local":
             raise NotImplementedError
 
-        logger.debug("Downloading '{}' to '{}'".format(from_info, to_info))
+        if self.isdir(from_info):
+            return self._download_dir(
+                from_info, to_info, name, no_progress_bar, file_mode, dir_mode
+            )
+        else:
+            return self._download_file(
+                from_info, to_info, name, no_progress_bar, file_mode, dir_mode
+            )
 
-        name = name or to_info.name
+    def _download_dir(
+        self, from_info, to_info, name, no_progress_bar, file_mode, dir_mode
+    ):
+        file_to_infos = (
+            to_info / file_to_info.relative_to(from_info)
+            for file_to_info in self.walk_files(from_info)
+        )
 
+        with ThreadPoolExecutor(max_workers=self.JOBS) as executor:
+            file_from_info = list(self.walk_files(from_info))
+            download_files = partial(
+                self._download_file,
+                name=name,
+                no_progress_bar=True,
+                file_mode=file_mode,
+                dir_mode=dir_mode,
+            )
+            with Tqdm(
+                file_from_info,
+                total=len(file_from_info),
+                desc="Downloading directory",
```

Comment:
To respect the incoming no_progress_bar flag for the directory itself🙂 
```suggestion
                desc="Downloading directory",
                disable=no_progress_bar,
```

Reply:
done